stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -880,7 +880,7 @@ without storing the intermediate result of `A*y`. As for the two-argument
 [`dot(_,_)`](@ref), this acts recursively. Moreover, for complex vectors, the
 first vector is conjugated.
 """
-dot(x, A, y) = dot(x, A*y) # generic fallback for cases that are not covered by specialized methods
+dot(x, A, y) = dot(A'x, y) # generic fallback for cases that are not covered by specialized methods
```

Comment:
Yes, sorry about that. I didn't consider the aspects you mentioned, but was lead by some tests in the large BLAS-array case. In case we rank speed higher than intermediate memory allocations, we would want to use `dot(A'x, y)` instead of `dot(x, A*y)`. So I thought this might be generally a good idea for a fallback, but probably isn't. I'll revert this change.

Reply:
In the general case you can't really say *a priori*` which is faster because `A` could be anything, from a row-major array to a linear operator in ApproxFun.

`A'*x` is probably faster than `A*x` for dense column-major `A`, i.e. `Array`.   However, in that case we are not using the fallback anyway, right?

Reply:
No, we don't. You're absolutely right. I got carried away during my efforts to speed up large dense matrix dot, which have failed miserably. :-(